src/syrupy/assertion.py

```diff
@@ -10,12 +13,20 @@
 
 from .exceptions import SnapshotDoesNotExist
 from .types import SerializableData
+from .utils import walk_snapshot_dir
 
 
 if TYPE_CHECKING:
-    from .serializers.base import AbstractSnapshotSerializer
     from .location import TestLocation
+    from .serializers.base import AbstractSnapshotSerializer
     from .session import SnapshotSession
+    from .types import SnapshotFiles
+
+
+AssertionResult = namedtuple(
+    "AssertionResult",
+    ["asserted", "created", "file", "name", "recalled", "success", "updated"],
```

Comment:
Preference for attrs: http://www.attrs.org/en/stable/why.html#namedtuples

Allows us to be stricter with types.

Reply:
mypy does not seem to play nice with `attr.make_class`

Reply:
What doesn't work? Can't you just use the attr.s decorator?

Reply:
Yeah but it's more verbose 😅
```py
AssertionResult = attr.make_class(
```
vs
```py
@attr.s
class AssertionResult(object):
    asserted = attr.ib()
    created = attr.ib()
    ...
```